habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -261,39 +276,69 @@ def reconfigure(self, config: Config):
                 ao: ao.joint_positions for ao in self.art_objs
             }
 
+    def set_robot_base_to_random_point(self):
+        for _ in range(50):
+            start_pos = self.pathfinder.get_random_navigable_point()
+
+            start_pos = self.safe_snap_point(start_pos)
+            start_rot = np.random.uniform(0, 2 * np.pi)
+
+            self.robot.base_pos = start_pos
+            self.robot.base_rot = start_rot
+
+            self.internal_step(-1)
```

Comment:
Instead of stepping the simulator, should we use discrete collision detection? It would be faster and less chance of messing up the state of other objects during this sampling procedure.